va/va_prot.h

```diff
@@ -0,0 +1,507 @@
+ * attribs[2].type = VAConfigAttribProtectedContentCipherMode;
+ * attribs[3].type = VAConfigAttribProtectedContentCipherSampleType;
+ * attribs[4].type = VAConfigAttribProtectedContentUsage;
+ * vaGetConfigAttributes(va_dpy, VAProfileProtected,
+ *     VAEntrypointProtectedContent, attribs, 5);
+ * if ((attribs[1].value & VA_PC_CIPHER_AES) == 0) {
+ *     // not find desired cipher algorithm
+ *     assert(0);
+ * }
+ * if ((attribs[2].value & VA_PC_BLOCK_SIZE_128) == 0) {
+ *     // not find desired block size
+ *     assert(0);
+ * }
+ * if ((attribs[3].value & VA_PC_CIPHER_MODE_CBC) == 0) {
+ *     // not find desired counter mode
+ *     assert(0);
+ * }
+ * if ((attribs[4].value & VA_PC_SAMPLE_TYPE_SUBSAMPLE) == 0) {
+ *     // not find desired sample type
+ *     assert(0);
+ * }
+ * if ((attribs[5].value & VA_PC_USAGE_WIDEVINE) == 0) {
+ *     // not find desired usage
+ *     assert(0);
+ * }
+ * \endcode
+ *
+ * \section api_pc_setup Set up a protected content session
+ *
+ * TEE Communication Entrypoint
+ * The protected content session provides a TEE session that is used to extract 
+ * TEE information. This information could be used to peform TEE operations. 
+
+ * Protected Content Entrypoint
+ * The protected content session can be attached to VA decode/encode/vp context
+ * to do decryption/protection in the pipeline.
+ * Before creating a protected content session, it needs to create a config
+ * first via vaCreateConfig(). Then using this config id to create a protected
+ * content session via vaCreateProtectedSession().
+ *
+ * The general control flow is demonstrated by the following pseudo-code:
+ * \code
+ * // Create config
+ * VAConfigID config_id;
+ *
+ * attribs[0].value = VA_PC_CIPHER_AES;
+ * attribs[1].value = VA_PC_BLOCK_SIZE_128;
+ * attribs[2].value = VA_PC_CIPHER_MODE_CBC;
+ * attribs[3].value = VA_PC_SAMPLE_TYPE_SUBSAMPLE;
+ * attribs[4].value = VA_PC_USAGE_WIDEVINE;
+ * va_status = vaCreateConfig(va_dpy, VAProfileProtected,
+ *     VAEntrypointProtectedContent, attribs, 5, &config_id);
+ * CHECK_VASTATUS(va_status, "vaCreateConfig");
+ * \endcode
+ *
+ * Once the config is set up, we can create protected content session via
+ vaCreateProtectedSession().
+ * \code
+ * // Create a protected session
+ * VAProtectedSessionID crypto_session;
+ *
+ * va_status = vaCreateProtectedSession(va_dpy, config_id, &crypto_session);
+ * CHECK_VASTATUS(va_status, "vaCreateProtectedSession");
+ * \endcode
+ *
+ * \section api_pc_exec TEE communication via vaProtectedSessionExecute()
+ *
+ * TEE Communication Entrypoint
+ * App needs to communicate with TEE to get TEE information or prime TEE with 
+ * information that will be utilized for future TEE operations/tasks. 
+ * 
+ * Protected Content Entrypoint
+ * Before starting decryption/encryption operation in GPU, app may need to communicate 
+ * with TEE to get encrypted assets for priming HWDRM pipeline for decryption. App need 
```

Comment:
what means "priming"?

Reply:
Priming is used to refer various types of initializations. For example, if license acquisition is being performed then priming means that TEE is already provisioned aka TEE has some sort of "cryptographic" whitelist of servers that TEE will use to do license acquisition for video playback. If HWDRM video playback is being performed then priming means that HWDRM eco-system TEE/GPU/Display has proper keys to do proper video playback etc.

Reply:
Put this into va_prot.h along with other text, please.

Reply:
Updated.